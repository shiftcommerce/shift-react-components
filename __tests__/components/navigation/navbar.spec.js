// Libraries
import React from 'react'
import { Provider } from 'react-redux'
import { createMockStore } from 'redux-test-utils'

// Components
import NavBar from '../../../src/components/navigation/navbar'
import Loading from '../../../src/objects/loading'

// Fixtures
import menu from '../../fixtures/menu'
import menuError from '../../fixtures/menu-error'
import menuLoading from '../../fixtures/menu-loading'

describe('The navbar', () => {
  test('renders the nav bar options', () => {
    // Act
    const wrapper = shallow(
      <Provider store={createMockStore()} >
        <NavBar menu={menu} />
      </Provider>
    )

    // Assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper).toContainReact(<NavBar menu={menu} />)
  })

  test('renders the error message', () => {
    // Act
    const wrapper = mount(
      <Provider store={createMockStore()} >
        <NavBar menu={menuError} />
      </Provider>
    )

    // Assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper).toIncludeText('Sorry! There is an error in loading menus')
  })

  test('renders loading spinner', () => {
    // Act
    const wrapper = mount(
      <Provider store={createMockStore()} >
        <NavBar menu={menuLoading} />
      </Provider>
    )

    // Assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper).toContainReact(<Loading />)
  })
})
