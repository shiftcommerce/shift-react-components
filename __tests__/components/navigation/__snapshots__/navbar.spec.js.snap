// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The navbar renders loading spinner 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
>
  <NavBar
    menu={
      Object {
        "data": Array [],
        "error": false,
        "loading": true,
      }
    }
  >
    <Loading>
      <Loading
        className="o-loading"
        color="#f5c6cb"
        delay={250}
        height={50}
        type="spin"
        width={50}
      >
        <div
          className="o-loading"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg class=\\"icon-blank\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\"></svg>
",
            }
          }
          style={
            Object {
              "fill": "#f5c6cb",
              "height": 50,
              "width": 50,
            }
          }
        />
      </Loading>
    </Loading>
  </NavBar>
</Provider>
`;

exports[`The navbar renders the error message 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
>
  <NavBar
    menu={
      Object {
        "data": Array [],
        "error": true,
        "loading": false,
      }
    }
  >
    <p
      className="c-nav__error"
    >
      Sorry! There is an error in loading menus 
    </p>
  </NavBar>
</Provider>
`;

exports[`The navbar renders the nav bar options 1`] = `
<NavBar
  menu={
    Array [
      Object {
        "error": false,
        "id": "14",
        "loading": false,
        "menu_items": Array [
          Object {
            "canonical_path": "/trays",
            "id": "54",
            "meta_attributes": Object {},
            "published": true,
            "title": "Trays",
          },
          Object {
            "canonical_path": "/nonslip-trays",
            "id": "55",
            "meta_attributes": Object {},
            "published": true,
            "title": "Nonslip trays",
          },
          Object {
            "canonical_path": "/jugs",
            "id": "56",
            "meta_attributes": Object {},
            "published": true,
            "title": "Jugs",
          },
          Object {
            "canonical_path": "/filters",
            "id": "57",
            "meta_attributes": Object {},
            "published": true,
            "title": "Filters",
          },
          Object {
            "canonical_path": "/tea",
            "id": "58",
            "meta_attributes": Object {},
            "published": true,
            "title": "Tea",
          },
          Object {
            "canonical_path": "/saucers",
            "id": "60",
            "meta_attributes": Object {},
            "published": true,
            "title": "Saucers",
          },
          Object {
            "canonical_path": "/cups",
            "id": "59",
            "meta_attributes": Object {},
            "published": true,
            "title": "Cups",
          },
          Object {
            "canonical_path": "/coffee",
            "id": "61",
            "meta_attributes": Object {},
            "published": true,
            "title": "Coffee",
          },
          Object {
            "canonical_path": "/servers",
            "id": "63",
            "meta_attributes": Object {},
            "published": true,
            "title": "Servers",
          },
          Object {
            "canonical_path": "/mugs",
            "id": "62",
            "meta_attributes": Object {},
            "published": true,
            "title": "Mugs",
          },
        ],
        "pagination": Object {
          "first": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
          "last": "/reference/v1/menus.json_api?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items&page%5Bnumber%5D=1&page%5Bsize%5D=25",
          "page_count": 1,
          "self": "/reference/v1/menus?fields%5Bmenu_items%5D=title%2Cslug%2Cmenu_items%2Citem%2Cbackground_image_link%2Cbackground_image%2Cpublished%2Ccanonical_path%2Cmeta_attributes&fields%5Bmenus%5D=title%2Creference%2Cupdated_at%2Cmenu_items&filter%5Bfilter%5D%5Breference%5D%5Beq%5D=mega-menu&include=menu_items",
          "total_entries": 1,
        },
        "reference": "mega-menu",
        "title": "Main Menu",
        "updated_at": "2018-11-16T15:19:36Z",
      },
    ]
  }
/>
`;
